packages/dashboard/src/managers/rmf-health-state-manager.ts

```diff
@@ -0,0 +1,74 @@
+import { DoorSummary, LiftSummary, DispenserSummary, RobotSummary } from 'react-components';
+
+export interface HealthStatus {
```

Comment:
I feel the design of this could be improved, there are several problem with this

1. `ItemSummaryState.outOfOrder` seems redundant, wouldn't it be equal to `spoiltDoors.length` etc?
2. I don't think the use of `ItemSummary.item` is required, the categories are already differentiatied based on different props (door, lifts etc).
3. `ItemSummary.itemSummary` and `ItemSummary.robotSummary` are optional, and `RobotSummaryState` is a subclass of `ItemSummaryState`. This make things unnecessarily confusing and breaks many advantages of type checking. Since doors etc and robots are already differentiated by different props, it's better to just let have use different interface, e.g.
```ts
export interface RobotSummary {
  spoiltRobots: SpoiltRobot[];
  summary: RobotSummaryState;
}
```